File: software/examples/delphi/ExampleSimple.pas
program ExampleSimple;

{$ifdef MSWINDOWS}{$apptype CONSOLE}{$endif}
{$ifdef FPC}{$mode OBJFPC}{$H+}{$endif}

uses
  SysUtils, IPConnection, BrickletTemperatureIR;

type
  TExample = class
  private
    ipcon: TIPConnection;
    tir: TBrickletTemperatureIR;
  public
    procedure Execute;
  end;

const
  HOST = 'localhost';
  PORT = 4223;
  UID = '365'; { Change to your UID }

var
  e: TExample;

procedure TExample.Execute;
var obj, amb: smallint;
begin
  { Create IP connection }
  ipcon := TIPConnection.Create;

  { Create device object }
  tir := TBrickletTemperatureIR.Create(UID, ipcon);

  { Connect to brickd }
  ipcon.Connect(HOST, PORT);
  { Don't use device before ipcon is connected }

  { Get current ambient and object temperatures (unit is °C/10) }
  obj := tir.GetObjectTemperature;
  amb := tir.GetAmbientTemperature;

  WriteLn(Format('Object Temperature: %f °C', [obj/10.0]));
  WriteLn(Format('Ambient Temperature: %f °C', [amb/10.0]));

  WriteLn('Press key to exit');
  ReadLn;
  ipcon.Destroy; { Calls ipcon.Disconnect internally }
end;

begin
  e := TExample.Create;
  e.Execute;
  e.Destroy;
end.

File: software/examples/vbnet/ExampleSimple.vb
Imports Tinkerforge

Module ExampleSimple
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "aNt" ' Change to your UID

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim tir As New BrickletTemperatureIR(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        ' Get current object and ambient temperatures (unit is °C/10)
        Dim obj As Short = tir.GetObjectTemperature()
        Dim amb As Short = tir.GetAmbientTemperature()

        System.Console.WriteLine("Object Temperature: " + (obj/10.0).ToString() + " C")
        System.Console.WriteLine("Ambient Temperature: " + (amb/10.0).ToString() + " C")

        System.Console.WriteLine("Press key to exit")
        System.Console.ReadKey()
        ipcon.Disconnect()
    End Sub
End Module
